### DISCUSSION TITLE
How to search for strings containing a specific word

### DISCUSSION THREAD

**Discussion by Geeyun-JY3** [WeblateOrg/weblate#7095] [Q&A] [answered]
For example, search for strings containg word “per”. 
Regular expressions (`r"\bper\b"`) does not seem to work: https://hosted.weblate.org/search/weblate/?offset=1&q=source%3Ar%22%5Cbper%5Cb%22&sort_by=-priority%2Cposition&checksum=

**Comment by nijel** [comment, marked as answer]
`\b` means backspace in PostgreSQL regular expressions, see https://www.postgresql.org/docs/current/functions-matching.html, so it does search for something else than you want.

There are more issues with complex regular expressions - we do validate them in Python, but they get executed by the database engine, see https://github.com/WeblateOrg/weblate/issues/5322.

**Reply by Geeyun-JY3** [reply]
So is possible to search for strings containing a specific word in a way now?

**Reply by nijel** [reply]
`target:r"(^|\W)per(\W|$)"` should do it

**Reply by Geeyun-JY3** [reply]
It works. Thank you so much.
